pathlib.Path with UNC paths throwing exception

**Describe the bug**
When creating fake UNC directory using `pathlib.Path` the `Path.mkdir` throws `FileNotFoundError` exception. When using simple windows paths like `C:/test/test1` each of following examples work.

**How To Reproduce**
```python
import os
import unittest
from pathlib import Path

from pyfakefs.fake_filesystem_unittest import TestCase


class BugTestCase(TestCase):
    def setUp(self) -> None:
        self.setUpPyfakefs()

    def test_os_makedirs(self):
        os.makedirs(r'\\test\folder1\folder2', exist_ok=True)
        self.assertTrue(Path(r'\\test\folder1\folder2').exists())

    def test_path_mkdir(self):
        Path(r'\\test\folder1\folder2').mkdir(parents=True, exist_ok=True)
        self.assertTrue(Path(r'\\test\folder1\folder2').exists())

    def test_continuous(self):
        Path(r'\\test').mkdir(parents=True, exist_ok=True)
        Path(r'\\test\folder1').mkdir(parents=True, exist_ok=True)
        Path(r'\\test\folder1\folder2').mkdir(parents=True, exist_ok=True)
        self.assertTrue(Path(r'\\test\folder1\folder2').exists())

    def test_fs_create_dir(self):
        self.fs.create_dir(r'\\test\folder1\folder2')
        self.assertTrue(Path(r'\\test\folder1\folder2').exists())


if __name__ == '__main__':
    unittest.main()
```

**Output**

```
============================= test session starts =============================
collecting ... collected 3 items

test_release.py::BugTestCase::test_fs_create_dir 
test_release.py::BugTestCase::test_os_makedirs 
test_release.py::BugTestCase::test_path_mkdir Clearing the cache


========================= 1 failed, 2 passed in 0.80s =========================
PASSED                  [ 33%]PASSED                    [ 66%]FAILED                     [100%]
test_release.py:15 (BugTestCase.test_path_mkdir)
self = WindowsPath('//test/folder1/folder2'), mode = 511, parents = True
exist_ok = True

    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
        """
        Create a new directory at this given path.
        """
        if self._closed:
            self._raise_closed()
        try:
>           self._accessor.mkdir(self, mode)

C:\Python38-64\lib\pathlib.py:1287: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\venv\lib\site-packages\pyfakefs\fake_pathlib.py:72: in _wrapped
    return strfunc(pathobj.filesystem, str(pathobj), *args, **kwargs)
..\venv\lib\site-packages\pyfakefs\fake_filesystem.py:2636: in makedir
    self.raise_os_error(errno.ENOENT, base_dir)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyfakefs.fake_filesystem.FakeFilesystem object at 0x0000024A491A8B50>
err_no = 2, filename = '\\\\test\\folder1\\', winerror = None

    def raise_os_error(
        self,
        err_no: int,
        filename: Optional[AnyString] = None,
        winerror: Optional[int] = None,
    ) -> NoReturn:
        """Raises OSError.
        The error message is constructed from the given error code and shall
        start with the error string issued in the real system.
        Note: this is not true under Windows if winerror is given - in this
        case a localized message specific to winerror will be shown in the
        real file system.
    
        Args:
            err_no: A numeric error code from the C variable errno.
            filename: The name of the affected file, if any.
            winerror: Windows only - the specific Windows error code.
        """
        message = os.strerror(err_no) + " in the fake filesystem"
        if winerror is not None and sys.platform == "win32" and self.is_windows_fs:
            raise OSError(err_no, message, filename, winerror)
>       raise OSError(err_no, message, filename)
E       FileNotFoundError: [Errno 2] No such file or directory in the fake filesystem: '\\\\test\\folder1\\'

..\venv\lib\site-packages\pyfakefs\fake_filesystem.py:428: FileNotFoundError

During handling of the above exception, another exception occurred:

self = <tests.test_release.BugTestCase testMethod=test_path_mkdir>

    def test_path_mkdir(self):
>       Path(r'\\test\folder1\folder2').mkdir(parents=True, exist_ok=True)

test_release.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Python38-64\lib\pathlib.py:1291: in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
C:\Python38-64\lib\pathlib.py:1287: in mkdir
    self._accessor.mkdir(self, mode)
..\venv\lib\site-packages\pyfakefs\fake_pathlib.py:72: in _wrapped
    return strfunc(pathobj.filesystem, str(pathobj), *args, **kwargs)
..\venv\lib\site-packages\pyfakefs\fake_filesystem.py:2636: in makedir
    self.raise_os_error(errno.ENOENT, base_dir)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyfakefs.fake_filesystem.FakeFilesystem object at 0x0000024A491A8B50>
err_no = 2, filename = '\\\\test\\folder1', winerror = None

    def raise_os_error(
        self,
        err_no: int,
        filename: Optional[AnyString] = None,
        winerror: Optional[int] = None,
    ) -> NoReturn:
        """Raises OSError.
        The error message is constructed from the given error code and shall
        start with the error string issued in the real system.
        Note: this is not true under Windows if winerror is given - in this
        case a localized message specific to winerror will be shown in the
        real file system.
    
        Args:
            err_no: A numeric error code from the C variable errno.
            filename: The name of the affected file, if any.
            winerror: Windows only - the specific Windows error code.
        """
        message = os.strerror(err_no) + " in the fake filesystem"
        if winerror is not None and sys.platform == "win32" and self.is_windows_fs:
            raise OSError(err_no, message, filename, winerror)
>       raise OSError(err_no, message, filename)
E       FileNotFoundError: [Errno 2] No such file or directory in the fake filesystem: '\\\\test\\folder1'

..\venv\lib\site-packages\pyfakefs\fake_filesystem.py:428: FileNotFoundError

Process finished with exit code 1
```

**Your environment**
```bash
Windows-10-10.0.19041-SP0
Python 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
pyfakefs 5.2.4
pytest 7.4.0
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -6,6 +6,8 @@ The released versions correspond to PyPI releases.
 ### Fixes
 * removed a leftover debug print statement (see [#869](../../issues/869))
 * make sure tests work without HOME environment set (see [#870](../../issues/870))
+* automount drive or UNC path under Windows if needed for `pathlib.Path.mkdir()`
+  (see [#890](../../issues/890))
 
 ## [Version 5.2.3](https://pypi.python.org/pypi/pyfakefs/5.2.3) (2023-08-18)
 Fixes a rare problem on pytest shutdown.

--- a/pyfakefs/fake_filesystem.py
+++ b/pyfakefs/fake_filesystem.py
@@ -2642,12 +2642,16 @@ class FakeFilesystem:
 
         if self.is_windows_fs:
             dir_name = self.absnormpath(dir_name)
-        parent_dir, _ = self.splitpath(dir_name)
+        parent_dir, rest = self.splitpath(dir_name)
         if parent_dir:
             base_dir = self.normpath(parent_dir)
             ellipsis = matching_string(parent_dir, self.path_separator + "..")
             if parent_dir.endswith(ellipsis) and not self.is_windows_fs:
                 base_dir, dummy_dotdot, _ = parent_dir.partition(ellipsis)
+            if self.is_windows_fs and not rest and not self.exists(base_dir):
+                # under Windows, the parent dir may be a drive or UNC path
+                # which has to be mounted
+                self._auto_mount_drive_if_needed(parent_dir)
             if not self.exists(base_dir):
                 self.raise_os_error(errno.ENOENT, base_dir)
 

--- a/pyfakefs/tests/fake_pathlib_test.py
+++ b/pyfakefs/tests/fake_pathlib_test.py
@@ -706,6 +706,18 @@ class FakePathlibPathFileOperationTest(RealPathlibTestCase):
             errno.EEXIST, self.path(file_name).mkdir, exist_ok=True
         )
 
+    @unittest.skipIf(not is_windows, "Windows specific behavior")
+    def test_mkdir_with_automount_unc_path(self):
+        self.skip_real_fs()
+        self.path(r"\\test\unc\foo").mkdir(parents=True)
+        self.assertTrue(self.path(r"\\test\unc\foo").exists())
+
+    @unittest.skipIf(not is_windows, "Windows specific behavior")
+    def test_mkdir_with_automount_drive(self):
+        self.skip_real_fs()
+        self.path(r"d:\foo\bar").mkdir(parents=True)
+        self.assertTrue(self.path(r"d:\foo\bar").exists())
+
     def test_rmdir(self):
         dir_name = self.make_path("foo", "bar")
         self.create_dir(dir_name)

